### PR TITLE
GROW-170 Cause selector component

### PR DESCRIPTION
**Pull request by ryan-ludwig**
*Screenshots from Aaron's designs below*

Component needs to be able to be used as a radio:
image

A checkbox:
image

And an icon only
image

### CODE REVIEW

**Comment by ryan-ludwig** [comment]
I went back and forth a lot over how to structure this component. Should it just be the icon image? Maybe we have a few components instead of 1? Should it be a KvComponent? I'm still open to ways to structure this better.

**Comment by mcstover** [comment]
@ryan-ludwig I think it's fine for this to be specialized as you've done it. It will most likely only live in 1-2 specialized contexts anyhow.

File: .storybook/stories/KvCauseSelector.stories.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
+import KvCauseSelector from '@/components/Kv/KvCauseSelector';
+
+export const causeList = [
```
Comment (eddieferrer) [inline .storybook/stories/KvCauseSelector.stories.js:3]: If you don't export this, it won't be picked up as a story and you don't have to exclude it below. It should just be a global variable in this file.

File: src/components/Kv/KvCauseSelector.vue [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+import { paramCase } from 'change-case';
+import KvIcon from '@/components/Kv/KvIcon';
+
+const imageRequire = require.context('@/assets/images/cause-selector/', true);
+
+export default {
+	components: {
+		KvIcon
+	},
+	model: {
```
Comment (eddieferrer) [inline src/components/Kv/KvCauseSelector.vue:64]: Why do we need this here? I think this only applies if you are using v-model directly

File: .storybook/stories/KvCauseSelector.stories.js (hunk 1 above)
Comment (ryan-ludwig) [reply .storybook/stories/KvCauseSelector.stories.js:3]: Oh, I'm using it in a WIP story for KvCarousel right now

Comment (eddieferrer) [reply .storybook/stories/KvCauseSelector.stories.js:3]: I see, if you're going to be using it in multiple stories, then maybe lets add it as a separate file in `mock-data`

File: src/components/Kv/KvCauseSelector.vue (hunk 2 above)
Comment (ryan-ludwig) [reply src/components/Kv/KvCauseSelector.vue:64]: Good catch, left over from an earlier iteration.

**Pull request merged**